modules/core/src/ocl.cpp

```diff
@@ -5129,4 +5228,175 @@ bool internal::isCLBuffer(UMat& u)
     return true;
 }
 
+struct Timer::Impl
+{
+    const Queue queue;
+
+    Impl(const Queue& q)
+        : queue(q)
+        , initted_(false)
+        , running_(false)
+        , has_run_at_least_once_(false)
+    {
+        init();
+    }
+
+    ~Impl()
+    {
+        clWaitForEvents(1, &start_gpu_cl_);
+        clWaitForEvents(1, &stop_gpu_cl_);
+        clReleaseEvent(start_gpu_cl_);
+        clReleaseEvent(stop_gpu_cl_);
+    }
+
+    void start()
+    {
+#ifdef HAVE_OPENCL
+        if (!running())
+        {
+            clWaitForEvents(1, &start_gpu_cl_);
+            clReleaseEvent(start_gpu_cl_);
+            ocl::Kernel kernel("null_kernel_float", ocl::core::benchmark_oclsrc);
+            float arg = 0;
+            clSetKernelArg((cl_kernel)kernel.ptr(), 0, sizeof(arg), &arg);
+            clEnqueueTask((cl_command_queue)queue.ptr(), (cl_kernel)kernel.ptr(), 0,
+                          NULL, &start_gpu_cl_);
```

Comment:
Did you try to use events from [clEnqueueMarker()](https://www.khronos.org/registry/OpenCL/sdk/1.1/docs/man/xhtml/clEnqueueMarker.html) (OpenCL 1.1) or preferable [clEnqueueBarrierWithWaitList()](https://www.khronos.org/registry/OpenCL/sdk/1.2/docs/man/xhtml/clEnqueueBarrierWithWaitList.html) (OpenCL 1.2+) instead of kernel task submitting?

Reply:
Thanks for your advice, I will try this in next commit